src/common/types.lua

```diff
@@ -0,0 +1,52 @@
+-- need to specify this as a module so it can be imported
+local types = {}
```

Comment:
What exactly is getting exported here? In the other repo, declaring the type/alias anywhere allows linting to use it anywhere without imports. Can you clarify where you got this pattern from and if it's necessary?